src/xpk/commands/info.py

```diff
@@ -0,0 +1,219 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ..utils import xpk_exit, xpk_print
+from ..core.kueue import verify_kueuectl_installation
+from .cluster import set_cluster_command
+from ..core.commands import (
+    run_command_for_value,
+)
+from ..core.core import (
+    add_zone_and_project,
+)
+import json
+from tabulate import tabulate
+from typing import Optional
+from argparse import Namespace
+
+table_fmt = 'plain'
+
+
+def prepare_kueuectl(args: Namespace) -> None:
+  """Verify if kueuectl is installed.
+  Args:
+    args: user provided arguments.
+  Returns:
+    None
+  """
+  xpk_print('Veryfing kueuectl installation')
+
+  verify_kueuectl_installed_code = verify_kueuectl_installation(args)
+  if verify_kueuectl_installed_code == 0:
+    xpk_print('kueuectl installed')
+
+  if verify_kueuectl_installed_code != 0:
+    xpk_print(
+        'kueuectl not installed. Please follow'
+        ' https://kueue.sigs.k8s.io/docs/reference/kubectl-kueue/installation/'
+        ' to install kueuectl.'
+    )
+    xpk_exit(verify_kueuectl_installed_code)
+
+
+def info(args: Namespace) -> None:
+  """Provide info about localqueues, clusterqueues and their resources.
+
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    None
+  """
+  add_zone_and_project(args)
+
+  set_cluster_command_code = set_cluster_command(args)
+  if set_cluster_command_code != 0:
+    xpk_exit(set_cluster_command_code)
+
+  apply_shared_flags(args)
+
+  prepare_kueuectl(args)
+
+  lqs = run_kueuectl_list_localqueue(args)
+  cqs = run_kueuectl_list_clusterqueue(args)
+
+  aggregate_results(cqs, lqs)
+
+
+def apply_shared_flags(args: Namespace) -> None:
```

Comment:
Use `add_zone_and_project()` instead

Reply:
add_zone_and_project only retrieve values from gcloud. If we want to set different values for invoking info command, we should apply gcloud commands. Correct me if I am wrong.

Reply:
`add_zone_and_project()` retrives these values only if they are not set by the user. In other words, if a user specifies these values, the method doesn't change them. However it comes handy if a user doesn't specify them. That way a user doesn't need to worry about setting these values in every xpk invocation.